Delete on plural endpoints

paullaffitte opened the pull request:
It implements deleteRecords, deleteCollections and deleteBuckets, with the same filtering as listRecords. #95

leplatrem commented on src/collection.ts:
```diff
@@ -596,6 +596,60 @@ export default class Collection {
+    const path = endpoint.record(this.bucket.name, this.name);
+    // if (options.at) {
+    //   return this.getSnapshot<T>(options.at);
+    // } else {
```
leftovers?

leplatrem commented on src/collection.ts:
```diff
@@ -596,6 +596,60 @@ export default class Collection {
+      headers: this._getHeaders(options),
+      retry: this._getRetry(options),
+    });
+    // }
```
same ;)

leplatrem commented on src/bucket.ts:
```diff
@@ -398,6 +398,33 @@ export default class Bucket {
+      retry: this._getRetry(options),
+    });
+  }
+
```
We should probably add `deleteGroups()` now that we're at this :)

leplatrem commented:
@paullaffitte do you need any help? Let us know :pray: !

paullaffitte commented:
@leplatrem Sorry I was on vacation, I think it's okay, maybe I'll need help for the tests but I'll let you know. I'll try to finish it this week-end.

paullaffitte commented:
I'm done with writing tests, but I can't seems to find how to properly run the integration tests.
`npm run test` gives me: "Uncaught Error: spawn pserve ENOENT"
and `TEST_KINTO_SERVER=localhost:8888/v1 npm run test`: "TypeError: Cannot read property 'flush' of undefined"

dstaley commented:
@paullaffitte You'll need to be in a python venv to run the integration tests. (Alternatively you can `pip install kinto kinto-attachment`, but using a venv is highly recommended)

The following should work with Python 3 on Linux:

```sh
$ python3 -m venv .venv
$ source .venv/bin/activate
(.venv) $ pip install kinto kinto-attachment
$ npm run test
```

paullaffitte commented:
That's very nice that there is some automated tests now. If they were there at the time I started the PR, it would be already merge maybe haha.

Anyway, sorry about the time I took to complete it, I was overwhelmed by work/studies/personal projects and I kind of forgot about this PR. I rebased my branch today and updated what should be updated. I also have tested it locally but I guess that the automated tests will show us if my work is good or not.

leplatrem commented on src/base.ts:
```diff
@@ -605,10 +607,14 @@ export default class KintoClientBase {
    *     Number of times to retry each request if the server responds
```
Add missing `method` no? With default value (cf. call from `paginateList`)

leplatrem commented on src/base.ts:
```diff
@@ -708,6 +715,101 @@ export default class KintoClientBase {
+  }
+
+  /**
+   * Delete some pages from a paginated list, following the `next-page`
```
I think it would be better phrased with something like `Delete multiple objects, following the pagination if the number of objects exceeds the page limit`

leplatrem commented on src/collection.ts:
```diff
@@ -613,6 +613,55 @@ export default class Collection {
+   *
+   * @see http://kinto.readthedocs.io/en/stable/api/1.x/filtering.html
+   *
+   * Paginating is done by passing a `limit` option, then calling the `next()`
```
...`The default server pagination size can be forced by passing ... `


Also, isn't it automatic? I don't think we should mention the `calling the next() method` no?

paullaffitte replied:
I'm not sure about this, but the `listRecords` function in the same file has a very similar documentation and is using the same `PaginatedParams` object as parameter. If I remove it I think we should also update the documentation of `listRecords`, for consistency sake.

paullaffitte commented:
> > That's very nice that there is some automated tests now. If they were there at the time I started the PR, it would be already merge maybe haha.
> 
> What do you mean? This repo had tests since its first commit :) https://github.com/Kinto/kinto-http.js/tree/87c681c82a

I maybe just didn't remember about it, I was pretty sure that I had to run the tests locally.

I will do the requested changes during this week.

paullaffitte commented:
Looks like the tests on firefox failed for session issue rather than because of my code (that I didn't change since last successful pipeline anyway, so it would makes sense).

dstaley commented:
All good now :) I'll review this over the next few days. Thanks again for contributing!

dstaley commented:
@paullaffitte Thank you so much for contributing, and hanging in there while we reviewed this. I really appreciate your patience and your effort in getting this ready to merge! :tada: :tada:
